app/ocr.py

```diff
@@ -350,61 +369,65 @@ def get_surrounding_sents(self, token: Token) -> Span:
 
         return surrounding
 
-    def get_protected_grounds(self) -> str:
+    def get_protected_grounds(self):
         """
-        • This will return the protected ground(s) of the applicant. Special
+        This will return the protected ground(s) of the applicant. Special
         checks are needed. Checking for keywords is not enough, as sometimes
         documents label laws that describe each protected ground. Examples
         are 'Purely Political Offense' and 'Real Id Act'.
         """
-        protected_grounds: List[str] = [
-            'race',
-            'religion',
-            'nationality',
-            'social',
-            'political',
+        
+        pattern = [
+        [{"LOWER": "race"}], 
+        [{"LOWER": "religion"}], # expand to check for list of religions
+        [{"LOWER": "nationality"}], # currently, phrase is pulled but out of context
+        [{"LOWER": "social"}, {"LOWER": "group"}], 
+        [{"LOWER": "political"}, {"LOWER": "opinion"}],
+        [{"LOWER": "political"}, {"LOWER": "offense"}],
+        [{"LOWER": "political"}],
         ]
 
-        pgs = []
-
-        similar_pg: Callable[[str, float], Union[str, None]]
-        similar_pg = similar_in_list(protected_grounds)
+        religions = ['christianity','christian','islam','atheist','hinduism',
+                     'buddihism','jewish','judaism','islamist','sunni','shia',
+                    'muslim','buddhist','atheists','jew','hindu', 'atheism']
 
-        for token in self.doc:
+        politicals = ['political opinion', 'political offense']
 
-            sent: str = token.sent.text.lower()
+        confirmed_matches = []
 
-            s: Union[str, None] = similar_pg(token.text.lower(), 0.9)
+        # create pattern for specified religions
+        for religion in religions:
+            pattern.append([{"LOWER": religion}])
 
-            if s == 'social':
-                next_word = self.doc[token.i + 1].text.lower()
-                if not similar(next_word, 'group', 0.95):
-                    continue
+        potential_grounds = similar(matcher_pattern=pattern, file=self.doc)
 
-            elif s == 'political':
-                next_word = self.doc[token.i + 1].text.lower()
-                if similar(next_word, 'offense', 0.95):
+        for match in potential_grounds:
+        # remove 'nationality act' from potential_grounds
+            if match.text.lower() == 'nationality':
+                if 'act' in match.sent.text.lower():

```

Comment:
Needs to also filter out references to relevant immigration laws, which we have a file for! immigration_laws.csv

Reply:
Do you have a link to that csv?